### ISSUE TITLE
[no-base-to-string] Error class is incorrectly reported as generating [Object object]

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#1749]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-base-to-string": "error"
  }
}
```

```TS
const fn = (myError: Error) => console.log(myError.toString());
```

**Expected Result**
No error.

**Actual Result**
> ESLint: 'myError will evaluate to '[Object object]' when stringified.(@typescript-eslint/no-base-to-string)

**Additional Info**

I assume part of this is because `Error` doesn't define a `toString` method, and so inherits it from `Object`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.24.0` |
| `@typescript-eslint/parser`        | `2.24.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.14.2.` |

**Comment by bradzacher** [comment]
yeah this is a bug in the typescript types.
https://github.com/microsoft/TypeScript/blob/2458c8a0168c8c67509a1524efca3fbf792dde5f/src/lib/es5.d.ts#L952-L964

Because the types don't define that the `Error` type has its own `toString` method, the rule doesn't know.
There's a few cases of this happening:
https://github.com/typescript-eslint/typescript-eslint/issues/1655#issuecomment-593639305

**Comment by bradzacher** [comment]
Merging into that issue
